.github/workflows/ci.yml

```diff
@@ -54,12 +57,15 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v2
+        with:
+          enable-cache: "true"
       - uses: actions/setup-python@v5
         with:
           python-version: "3.12"
-          cache: pip
       - name: Install dependencies
-        run: pip install -r requirements.txt
+        run: uv sync
```

Comment:
While it might be benefical to keep the install and run steps separated, I _think_ this step is not 100% needed as uv will do it automatically for you when needed.

Reply:
Removed the `setup-python` step since `uv sync` and `uv run` install python in the virtualenv.